File: PetMate-frontend/lib/features/pets/domain/entities/pet.dart
import '../../../core/enums/energy_level.dart';
import '../../../core/enums/pet_gender.dart';
import '../../../core/enums/pet_type.dart';

/// A pet owned by a user in the Domain layer.
class Pet {
  const Pet({
    required this.id,
    required this.ownerId,
    required this.name,
    required this.type,
    required this.gender,
    this.breed,
    required this.ageYears,
    this.energyLevel,
    this.bio,
    this.photos = const [],
    this.isVaccinated = false,
    this.isNeutered = false,
    this.createdAt,
  });

  final String id;
  final String ownerId;
  final String name;
  final PetType type;
  final PetGender gender;
  final String? breed;
  final int ageYears;
  final EnergyLevel? energyLevel;
  final String? bio;
  final List<String> photos;
  final bool isVaccinated;
  final bool isNeutered;
  final DateTime? createdAt;
}

File: pawmate-backend/src/main/java/com/pawmate/backend/entity/Pet.java
package com.pawmate.backend.entity;

import com.pawmate.backend.enums.EnergyLevel;
import com.pawmate.backend.enums.PetGender;
import com.pawmate.backend.enums.PetType;
import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "pets")
public class Pet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Size(max = 100)
    @Column(nullable = false, length = 100)
    private String name;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private PetType type;

    @Size(max = 100)
    @Column(length = 100)
    private String breed;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private PetGender gender;

    @Min(0)
    private Integer age;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EnergyLevel energyLevel;

    @Size(max = 1000)
    @Column(length = 1000)
    private String description;

    @Builder.Default
    @Column(nullable = false)
    private boolean vaccinated = false;

    @Builder.Default
    @Column(nullable = false)
    private boolean neutered = false;

    @Builder.Default
    @Column(nullable = false)
    private boolean active = false;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User owner;

    @Builder.Default
    @OneToMany(mappedBy = "pet")
    private List<PetPhoto> photos = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}